src/v/cloud_storage/topic_mount_handler.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "cloud_storage/remote.h"
+#include "cloud_storage/remote_label.h"
+#include "cloud_storage/remote_path_provider.h"
+#include "cloud_storage/topic_mount_manifest.h"
+#include "cluster/topic_configuration.h"
+#include "model/fundamental.h"
+#include "utils/retry_chain_node.h"
+
+namespace cloud_storage {
+
+enum class topic_mount_result {
+    mount_manifest_does_not_exist,
+    mount_manifest_not_deleted,
+    success
+};
+
+std::ostream& operator<<(std::ostream& o, const topic_mount_result& r);
+
+enum class topic_unmount_result { mount_manifest_not_created, success };
+
+std::ostream& operator<<(std::ostream& o, const topic_unmount_result& r);
+
+class topic_mount_handler {
+public:
+    topic_mount_handler(
+      const cloud_storage_clients::bucket_name& bucket, remote& remote);
+
+    // Perform the mounting process by deleting the topic mount manifest.
+    ss::future<topic_mount_result> mount_topic(
+      const remote_label& label,
+      const cluster::topic_configuration& topic_cfg,
+      retry_chain_node& parent);
+
+    // Perform the unmounting process by uploading the topic mount manifest.
+    ss::future<topic_unmount_result> unmount_topic(
+      const remote_label& label,
+      const cluster::topic_configuration& topic_cfg,
```

Comment:
Remote label is also a member of topic configuration -- if we're passing in a config, any reason to not use its label?

Reply:
Corrected function signature for `topic_mount_handler::mount/unmount_topic()`.

We now use `topic_cfg.properties.remote_label.value_or(remote_label{model::default_cluster_uuid})` for the `source_label` passed to `topic_mount_manifest`.